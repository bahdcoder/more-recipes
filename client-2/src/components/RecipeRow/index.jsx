import React from 'react';
import PropTypes from 'prop-types';
import RecipeCard from './../RecipeCard';

const RecipeRow = ({ recipes }) => (
  <div className="card-deck wow fadeIn">
    {recipes.map(recipe => (
      <div className="col-md-4 col-lg-4" key={recipe.id}>
        <RecipeCard recipe={recipe} />
      </div>
    ))}
  </div>
);

RecipeRow.propTypes = {
  recipes: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default RecipeRow;
