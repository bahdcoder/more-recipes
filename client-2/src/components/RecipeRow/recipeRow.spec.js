/* global expect */
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import RecipeRow from './index';
import mockRecipes from './../../mock/recipes.json';

describe('The RecipeRow component', () => {
  it('Should mount without crashing', () => {
    const tree = renderer.create((
      <MemoryRouter>
        <RecipeRow recipes={mockRecipes} />
      </MemoryRouter>
    )).toJSON();

    expect(tree.children.length).toBe(3);
    expect(tree).toMatchSnapshot();
  });
});
