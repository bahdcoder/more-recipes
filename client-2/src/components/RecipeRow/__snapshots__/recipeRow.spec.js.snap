// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The RecipeRow component Should mount without crashing 1`] = `
<div
  className="card-deck wow fadeIn"
>
  <div
    className="col-md-4 col-lg-4"
  >
    <div
      className="wow fadeIn card mb-3"
    >
      <div
        className="img-zoom"
      >
        <img
          alt=""
          className="card-img-top"
          src="https://res.cloudinary.com/bahdcoder/image/upload/v1512897507/fcvvnaqichwk0k0eak39.png"
          style={
            Object {
              "height": 170,
            }
          }
        />
      </div>
      <div
        className="card-body"
      >
        <h5
          className="card-title h6 text-center"
        >
          <a
            href="/recipe/e304eba1-9f87-4424-8c8c-af603976fd23"
            onClick={[Function]}
          >
            Mashed potatoes chips
          </a>
        </h5>
        <hr />
        <p
          className="text-sm mb-3"
        >
          <small>
            <span
              className="text-muted"
            >
              by 
            </span>
            <a
              className="header-color"
              href="/user/7b3da821-8a11-4238-919e-871c99b1e98f"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              Doctor Strange
            </a>
          </small>
          <span
            className="text-muted float-right"
          >
            <i
              className="ion ion-clock mr-2"
            />
            120
             min
          </span>
        </p>
        <p
          className="text-muted h4 text-center my-2"
        >
          <span
            className="mr-3 h5"
          >
            <i
              className="ion ion-recipe-action ion-happy-outline"
            />
            <span
              className="ml-3"
            >
              4k
            </span>
          </span>
          <span
            className="mr-3 h5"
          >
            <i
              className="ion ion-recipe-action ion-sad-outline"
            />
            <span
              className="ml-3"
            >
              2k
            </span>
          </span>
          <span
            className="mr-3 h5"
          >
            <i
              className="ion ion-recipe-action ion-ios-heart-outline"
            />
            <span
              className="ml-3"
            >
              32
            </span>
          </span>
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-md-4 col-lg-4"
  >
    <div
      className="wow fadeIn card mb-3"
    >
      <div
        className="img-zoom"
      >
        <img
          alt=""
          className="card-img-top"
          src="https://res.cloudinary.com/bahdcoder/image/upload/v1513105171/juduffnpsffx9wplysux.png"
          style={
            Object {
              "height": 170,
            }
          }
        />
      </div>
      <div
        className="card-body"
      >
        <h5
          className="card-title h6 text-center"
        >
          <a
            href="/recipe/e304eba1-9f87-4424-8c8c-af603976fd23"
            onClick={[Function]}
          >
            title
          </a>
        </h5>
        <hr />
        <p
          className="text-sm mb-3"
        >
          <small>
            <span
              className="text-muted"
            >
              by 
            </span>
            <a
              className="header-color"
              href="/user/7b3da821-8a11-4238-919e-871c99b1e98f"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              Doctor Strange
            </a>
          </small>
          <span
            className="text-muted float-right"
          >
            <i
              className="ion ion-clock mr-2"
            />
            32321
             min
          </span>
        </p>
        <p
          className="text-muted h4 text-center my-2"
        >
          <span
            className="mr-3 h5"
          >
            <i
              className="ion ion-recipe-action ion-happy-outline"
            />
            <span
              className="ml-3"
            >
              4k
            </span>
          </span>
          <span
            className="mr-3 h5"
          >
            <i
              className="ion ion-recipe-action ion-sad-outline"
            />
            <span
              className="ml-3"
            >
              2k
            </span>
          </span>
          <span
            className="mr-3 h5"
          >
            <i
              className="ion ion-recipe-action ion-ios-heart-outline"
            />
            <span
              className="ml-3"
            >
              32
            </span>
          </span>
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-md-4 col-lg-4"
  >
    <div
      className="wow fadeIn card mb-3"
    >
      <div
        className="img-zoom"
      >
        <img
          alt=""
          className="card-img-top"
          src="https://res.cloudinary.com/bahdcoder/image/upload/v1512897507/fcvvnaqichwk0k0eak39.png"
          style={
            Object {
              "height": 170,
            }
          }
        />
      </div>
      <div
        className="card-body"
      >
        <h5
          className="card-title h6 text-center"
        >
          <a
            href="/recipe/e304eba1-9f87-4424-8c8c-af603976fd23"
            onClick={[Function]}
          >
            Mashed potatoes chips
          </a>
        </h5>
        <hr />
        <p
          className="text-sm mb-3"
        >
          <small>
            <span
              className="text-muted"
            >
              by 
            </span>
            <a
              className="header-color"
              href="/user/7b3da821-8a11-4238-919e-871c99b1e98f"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              Doctor Strange
            </a>
          </small>
          <span
            className="text-muted float-right"
          >
            <i
              className="ion ion-clock mr-2"
            />
            120
             min
          </span>
        </p>
        <p
          className="text-muted h4 text-center my-2"
        >
          <span
            className="mr-3 h5"
          >
            <i
              className="ion ion-recipe-action ion-happy-outline"
            />
            <span
              className="ml-3"
            >
              4k
            </span>
          </span>
          <span
            className="mr-3 h5"
          >
            <i
              className="ion ion-recipe-action ion-sad-outline"
            />
            <span
              className="ml-3"
            >
              2k
            </span>
          </span>
          <span
            className="mr-3 h5"
          >
            <i
              className="ion ion-recipe-action ion-ios-heart-outline"
            />
            <span
              className="ml-3"
            >
              32
            </span>
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
